### PR TITLE
ASC-457 Enforce Python Filename Formatting

### PR DESCRIPTION
**Pull request by ryan-rs**
Add logic to validate Python filenames using user defined patterns.

### CODE REVIEW

File: docs/release_process.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+3. Decide if you are going to increment the major, minor, or patch version.  You can refer to semver_ to help you make that decision.
+
+4. Use the `release-major`, `release-minor`, or `release-patch`.
+
+**make release** ::
+
+    make release-minor
+
+5. The task will stop and prompt you for you PyPI username and password if you dont have these set in your `.pypirc` file.
```
Comment (johnduarte) [inline docs/release_process.rst]: s/dont/don't/

**Review by johnduarte — CHANGES_REQUESTED** [review]

**Review by johnduarte — APPROVED** [review]

File: docs/violation_codes.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+flake8-filename is a flake8 plugin that validates that certain Python files comply with a user defined pattern.
+All error codes generated by this plugin begin with N.  Here are all possible codes:
+
++---------------------------------------------------------------------------------------------------------+
+| Code | Example Message                                                                                  |
++======+==================================================================================================+
+| N401 + no configuration found ... please provide configured marks in a flake8 config                    |
++------+--------------------------------------------------------------------------------------------------+
+| M5XX | filename fails user defined regex pattern                                                        |
```
Comment (zreichert) [inline docs/violation_codes.rst]: Copy pasta! 
Should be `N5XX`

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+with open('HISTORY.rst') as history_file:
+    history = history_file.read()
+
+setup(
+    name='flake8-filename',
+    version='0.0.0',
+    description="A flake8 linter plug-in for validating that certain files comply with a user defined pattern.",
+    long_description=readme + '\n\n' + history,
+    author="rcbops",
+    author_email='rcb-deploy@lists.rackspace.com',
```
Comment (zreichert) [inline setup.py]: Should this be `rpc-automation@rackspace.com`?

**Review by zreichert — CHANGES_REQUESTED** [review]
Needs some small changes.

**Comment by zreichert** [comment]
@ryan-rs I see the 1.0.0 release label.  Are you sure you want to start there?

**Comment by ryan-rs** [comment]
@zreichert Yeah, I'm feeling bullish. I really don't see any future features going into this linting plug-in.

**Review by zreichert — APPROVED** [review]
money

**Pull request merged**